### PR TITLE
Added better color on selected text in different themes

### PR DESCRIPTION
**Pull request by rt4914**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Currently in **Dark Theme** on Android device when user selects any text the foreground:background contrast ratio is very low which makes the selected text difficult to read. This issue is in all html tags like h1,h2,p, etc.

To fix this, I have introduced `--selection-bg-color` and `--selection-color` in themes which are then getting used in `main.scss` file to fix this issue globally.

## QA Instructions, Screenshots, Recordings

**Web / Android - Light Theme**
Screenshot 2021-07-26 at 9 55 32 PM   Screenshot 2021-07-26 at 9 55 41 PM

**Web/Android - Dark Theme**
Screenshot 2021-07-26 at 9 56 13 PM   Screenshot 2021-07-26 at 9 56 19 PM

## Pending Work Items / Questions
1. There are currently 4 theme files (`hacker.scss`, `minimal.scss`, `night.scss`, `pink.scss`) so we need proper background and foreground selection color codes for each one of these themes.
2. Is the overall approach correct?
3. How can we write test cases for frontend tests? Are there any references/sample code?

### UI accessibility concerns?

Yes. The foreground:background color contrast ratio should be more than 4.5:1. [Reference](https://webaim.org/articles/contrast/evaluating)

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [x] I need help with writing tests (If there are any test cases)

## [Forem core team only] How will this change be communicated?

Not applicable.

- [ ] I've updated the [Developer Docs](https://docs.forem.com) and/or
      [Admin Guide](https://admin.forem.com/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] I've updated the README or added inline documentation
- [ ] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [ ] I will share this change internally with the appropriate teams
- [x] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

No

## [optional] What gif best describes this PR or how it makes you feel?

Happy Dance

### CODE REVIEW

**Comment by joshpuetz** [comment]
Tagging @aitchiss for her input!

**Comment by aitchiss** [comment]
I'd like to look into this one a little bit if that's OK - I agree with the motivation, but a lot of assistive tools allow users to customise things like selection style to meet their needs, and I'd like to make sure we don't accidentally break this behaviour by overriding it ourselves. I'm off for a week of holidays from tomorrow so it might be a while before I can feed in properly though 😅 

It's also worth noting that [`::selection` isn't available on Safari/Chrome on iOS](https://caniuse.com/css-selection).

> How can we write test cases for frontend tests? Are there any references/sample code?

I don't think we'd be able to write meaningful frontend tests for this kind of change, but you can find our docs with details on how we test the frontend here:

- [Component frontend tests](https://docs.forem.com/tests/frontend-tests/)
- [End to end tests](https://docs.forem.com/tests/e2e-tests/)

**Comment by aitchiss** [comment]
Sorry for the delay looking into this! I did a little bit of research, and to sum up what I found:

- There's a lot of ways that users can customise colours both on devices and in browsers. There's a [nice gov.uk article about it](https://accessibility.blog.gov.uk/2017/03/27/how-users-change-colours-on-websites/) that I found useful
- If we start controlling selection colour in the app, then a user's system preferences are no longer applied. For example, on Mac you have a couple of highlight colour features - in `Preferences > General` and in `Accessibility > Display > Invert colours`. In both cases the system user preference is ignored if we specify a different colour in CSS
- We also need to consider how link colours for each theme will hold up against the new selection colour. [This CSS Tricks article has a good tip for a Sass mixin to help with this](https://css-tricks.com/a-whole-bunch-of-places-to-consider-contrast-in-a-single-paragraph/)
- As @rt4914 has identified, the quality of contrast at the moment is pretty device-dependent. I see something slightly different on dark theme on my Pixel 5, but it's equally poor contrast 😅 

Screenshot_20210804-135629


Long story short, I'd say yes let's do this for each defined theme **_but_** let's use system defaults for the default theme. This way, if a user has custom highlight preferences set at a system level, they can still see them applied unless they decide to use one of the pre-defined themes. And when we pick out the colours, let's also make sure the links within paragraphs contrast enough with the background too ✨

**Comment by rt4914** [comment]
> Sorry for the delay looking into this! I did a little bit of research, and to sum up what I found:
> 
> * There's a lot of ways that users can customise colours both on devices and in browsers. There's a [nice gov.uk article about it](https://accessibility.blog.gov.uk/2017/03/27/how-users-change-colours-on-websites/) that I found useful
> * If we start controlling selection colour in the app, then a user's system preferences are no longer applied. For example, on Mac you have a couple of highlight colour features - in `Preferences > General` and in `Accessibility > Display > Invert colours`. In both cases the system user preference is ignored if we specify a different colour in CSS
> * We also need to consider how link colours for each theme will hold up against the new selection colour. [This CSS Tricks article has a good tip for a Sass mixin to help with this](https://css-tricks.com/a-whole-bunch-of-places-to-consider-contrast-in-a-single-paragraph/)
> * As @rt4914 has identified, the quality of contrast at the moment is pretty device-dependent. I see something slightly different on dark theme on my Pixel 5, but it's equally poor contrast 😅
> 
> Screenshot_20210804-135629
> 
> Long story short, I'd say yes let's do this for each defined theme **_but_** let's use system defaults for the default theme. This way, if a user has custom highlight preferences set at a system level, they can still see them applied unless they decide to use one of the pre-defined themes. And when we pick out the colours, let's also make sure the links within paragraphs contrast enough with the background too ✨

@aitchiss Great. Thanks for all the insights. I will prioritise this as soon as my initial architecture for android project is finalised.

**Comment by citizen428** [comment]
Hi @rt4914 👋 What's your plan for this PR, will continue working on it? 😊

**Comment by rt4914** [comment]
@citizen428 Apologies for the delay. For now I would like to discontinue from this issue as I want to wrap up the android project first. I will un-assign myself for now.

**Comment by citizen428** [comment]
No problem, thanks anyway! 😃
